From a86fb0b9e25f67c420c1109da6431cf34a9a0558 Mon Sep 17 00:00:00 2001
From: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
Date: Tue, 2 May 2017 17:44:15 -0700
Subject: [PATCH 0860/2345] usb: dwc3: dwc3-pci: Add support to enumerate Intel
 Dual Role phy

This patch adds support to enumerate Intel Dual Role phy driver for
Broxton platform.

Signed-off-by: Sathyanarayanan Kuppuswamy <sathyanarayanan.kuppuswamy@intel.com>
---
 drivers/usb/dwc3/dwc3-pci.c          | 21 +++++++++++++++++++++
 drivers/usb/phy/phy-intel-dualrole.c |  1 -
 2 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/dwc3-pci.c b/drivers/usb/dwc3/dwc3-pci.c
index 09c0454..4ca7cdb 100644
--- a/drivers/usb/dwc3/dwc3-pci.c
+++ b/drivers/usb/dwc3/dwc3-pci.c
@@ -57,6 +57,7 @@
  */
 struct dwc3_pci {
 	struct platform_device *dwc3;
+	struct platform_device	*usb2_phy;
 	struct pci_dev *pci;
 
 	guid_t guid;
@@ -125,6 +126,24 @@ static int dwc3_pci_quirks(struct dwc3_pci *dwc)
 			dwc->has_dsm_for_pm = true;
 		}
 
+		if (pdev->device == PCI_DEVICE_ID_INTEL_BXT ||
+		    pdev->device == PCI_DEVICE_ID_INTEL_BXT_M ||
+		    pdev->device == PCI_DEVICE_ID_INTEL_APL ) {
+
+			dwc->usb2_phy = platform_device_alloc(
+							"intel_usb_dr_phy", 0);
+			if (!dwc->usb2_phy)
+				return -ENOMEM;
+
+			dwc->usb2_phy->dev.parent = &pdev->dev;
+	                ret = platform_device_add(dwc->usb2_phy);
+		        if (ret) {
+			        platform_device_put(dwc->usb2_phy);
+				return ret;
+			}
+		}
+
+
 		if (pdev->device == PCI_DEVICE_ID_INTEL_BYT) {
 			struct gpio_desc *gpio;
 
@@ -246,6 +265,8 @@ static void dwc3_pci_remove(struct pci_dev *pci)
 
 	device_init_wakeup(&pci->dev, false);
 	pm_runtime_get(&pci->dev);
+	if(dwc->usb2_phy)
+		platform_device_unregister(dwc->usb2_phy);
 	platform_device_unregister(dwc->dwc3);
 }
 
diff --git a/drivers/usb/phy/phy-intel-dualrole.c b/drivers/usb/phy/phy-intel-dualrole.c
index f15d239..a49bd1e 100644
--- a/drivers/usb/phy/phy-intel-dualrole.c
+++ b/drivers/usb/phy/phy-intel-dualrole.c
@@ -51,7 +51,6 @@ static int intel_usb_mux_update(struct intel_dr_phy *intel_phy,
 {
 	struct usb_bus *host = intel_phy->phy.otg->host;
 	struct usb_gadget *gadget = intel_phy->phy.otg->gadget;
-	int ret;
 
 	if (!host || !gadget || !gadget->dev.parent || !intel_phy->edev)
 		return -ENODEV;
-- 
2.7.4

